js/assessment.js

```diff
@@ -225,10 +229,21 @@ define([
             if (this._assessments._byPageId[pageId] === undefined) {
                 this._assessments._byPageId[pageId] = [];
             }
+
             this._assessments._byPageId[pageId].push(assessmentModel);
 
-            if (assessmentId) {
-                this._assessments._byAssessmentId[assessmentId] = assessmentModel;
+            if (assessmentId !== undefined) {
+                if (assessmentId === '') {
+                    Adapt.log.warn("An assessment has been registered with an empty value for 'id'");
+                }
+
+                if (!this._assessments._byAssessmentId[assessmentId]) {
+                    this._assessments._byAssessmentId[assessmentId] = assessmentModel;
+                } else {
+                    Adapt.log.warn("An assessment with an id of '" + assessmentId + "' already exists!");
+                }
+            } else {
+                Adapt.log.warn("An assessment has been registered with an undefined value for 'id'");
             }
```

Comment:
Do we need two errors for undefined + empty string? This code is unnecessarily nested, what about:
```javascript
if (assessmentId && !this._assessments._byAssessmentId[assessmentId]) {
    this._assessments._byAssessmentId[assessmentId] = assessmentModel;
} else if (!assessmentId) {
    Adapt.log.warn("An assessment has been registered with an undefined or empty string value for '_id'");
} else if (this._assessments._byAssessmentId[assessmentId]) {
    Adapt.log.warn("An assessment with an id of '" + assessmentId + "' already exists!");
}
```